technic/machines/register/cables.lua

```diff
@@ -96,32 +96,20 @@ function technic.register_cable_plate(nodename, data)
 		def.node_box["connect_"..notconnects[i]] = nil
 		if i == 1 then
 			def.on_place = function(itemstack, placer, pointed_thing)
-				local pointed_thing_diff = vector.subtract(pointed_thing.above, pointed_thing.under)
-				local index = pointed_thing_diff.x + (pointed_thing_diff.y*2) + (pointed_thing_diff.z*3)
-				local num = index < 0 and -index + 3 or index
-				local crtl = placer:get_player_control()
-				if (crtl.aux1 or crtl.sneak) and not (crtl.aux1 and crtl.sneak) and index ~= 0 then
-					local fine_pointed = minetest.pointed_thing_to_face_pos(placer, pointed_thing)
-					fine_pointed = vector.subtract(fine_pointed, pointed_thing.above)
-					fine_pointed[xyz[index < 0 and -index or index]] = nil
-					local key_a, a = next(fine_pointed)
-					local key_b, b = next(fine_pointed, key_a)
-					local far_key = math.abs(a) > math.abs(b) and key_a or key_b
-					local far = fine_pointed[far_key]
-					-- Plate facing
-					-- X pair floor +X 4 -X 1 -> Z pair, Y pair
-					-- Y pair floor +Y 5 -Y 2 -> X pair, Z pair
-					-- Z pair floor +Z 6 -Z 3 -> X pair, Y pair
-					if math.abs(far) < 0.3 then
-						num = num < 4 and num + 3 or num - 3
-					elseif far_key == "x" then
-						num = far < 0 and 1 or 4
-					elseif far_key == "y" then
-						num = far < 0 and 2 or 5
-					else
-						num = far < 0 and 3 or 6
+				-- temporary bypass for "diagonal" above/under vectors (caused by oversized nodes)
+				-- see github.com/mt-mods/technic/pull/358
+				local axis_count = 0
+				for axis in pairs(xyz) do
+					axis_count = axis_count + (pointed_thing.under[axis] == pointed_thing.above[axis] and 0 or 1)
+					if axis_count > 1 then
+						return itemstack
 					end
 				end
+
+				local pointed_thing_diff = vector.direction(pointed_thing.above, pointed_thing.under)
+				local dir = math.abs(pointed_thing_diff.x + 2*pointed_thing_diff.y + 3*pointed_thing_diff.z) -- 1, 2 or 3
+				local num = dir + 3/2*(1+pointed_thing_diff[dir]) -- dir + additional offset
```

Comment:
What was this supposed to do and how exactly does it handle placement options?

For now it does fail significant portion of test cases and instead of fixing bug it seems to be an attempt to rework whole user facing placement behavior.

Reply:
* `pointed_thing_diff` : you know ;)
* `dir`: the previously index, but reinforced
  * x = +/- 1 returns 1
  * y = +/- 1 returns 2
  * z = +/-1 returns 3
* `num` : the number of the `cable_plate_num` model ; it's working like `y = ax+b`
  * `dir +` : because its keep that `b` which means the axis
  * `1+pointed_thing_diff[dir]` : 
    * returns 0 for x/y/z = -1
    * returns 2 for x/y/z = 1
  * `*3/2` : add this offset of 0 or 2, depending on the value

This is relatively short, and understandable, according to me.

Reply:
It is but it is reworking whole placement behavior which in my opinion is not good thing to do.

Needs some more discussion about before going there. Bugs can be fixed without changing behavior, for that I'd of course give approval but for changing behavior without good enough reasoning reject from me.